lib/polygeist/Ops.cpp

```diff
@@ -5778,6 +5778,10 @@ LogicalResult fixupGetFunc(LLVM::CallOp op, OpBuilder &rewriter,
 
   Value pval = op.getOperand(0);
 
+  // We can't get type info from opaque pointer which we now use TODO what do we
```

Comment:
@wsmoses 

Can you take a look at what we should be doing here instead?

Reply:
Most of the remaining failing tests are related to this

Reply:
in LLVM itself there is a getFunctionType from the CallInst.

we should use the MLIR-equivalent of that here to get FT

Reply:
LLVM: https://llvm.org/doxygen/InstrTypes_8h_source.html#l01270

MLIR: https://github.com/llvm/llvm-project/blob/b300f8c89829f920102e6ec5acc7d1b134bcff75/mlir/include/mlir/Dialect/LLVMIR/LLVMOps.td#L628

I don't immediately see anything upstream MLIR that does the LLVM similar. @ftynse if you know offhand

If not we may need to add support to upstream MLIR for

Reply:
I think the assumption so far was that functions are non-variadic so the function type can be trivially reconstructed from the lists of operands and results of the call op. For variadic functions, there's a FIXME just above: https://github.com/llvm/llvm-project/blob/b300f8c89829f920102e6ec5acc7d1b134bcff75/mlir/include/mlir/Dialect/LLVMIR/LLVMOps.td#L584-L586